flink-runtime/src/main/java/org/apache/flink/streaming/runtime/tasks/mailbox/TaskMailbox.java

```diff
@@ -138,21 +144,23 @@ public interface TaskMailbox {
      *
      * @return true if there is at least one element in the batch; that is, if there is any mail at
      *     all at the time of the invocation.
+     * @throws IllegalStateException if called from non-mailbox thread.
      */
     boolean createBatch();
 
     /**
-     * Returns an optional with either the oldest mail from the batch (head of queue) if the batch
+     * Returns an optional with either the oldest mail from the batch (head of batch) if the batch
```

Comment:
`(head of queue)` -> `(head of batch)`. We only fetch the head mail from the `batch` not the `queue` with `tryTakeFromBatch`.